exporter/sentryexporter/sentry_exporter.go

```diff
@@ -0,0 +1,395 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sentryexporter
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/getsentry/sentry-go"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"go.opentelemetry.io/collector/translator/conventions"
+)
+
+const (
+	sentryStatusUnknown       = "unknown"
+	otelSentryExporterVersion = "0.0.1"
+	otelSentryExporterName    = "sentry.opentelemetry"
+)
+
+// canonicalCodes maps OpenTelemetry span codes to Sentry's span status.
+// See numeric codes in https://godoc.org/github.com/open-telemetry/opentelemetry-proto/gen/go/trace/v1#Status_StatusCode.
+var canonicalCodes = [...]string{
+	"ok",
+	"cancelled",
+	sentryStatusUnknown,
+	"invalid_argument",
+	"deadline_exceeded",
+	"not_found",
+	"already_exists",
+	"permission_denied",
+	"resource_exhausted",
+	"failed_precondition",
+	"aborted",
+	"out_of_range",
+	"unimplemented",
+	"internal",
+	"unavailable",
+	"data_loss",
+	"unauthenticated",
+}
+
+// SentryExporter defines the Sentry Exporter.
+type SentryExporter struct {
+	transport transport
+}
+
+// pushTraceData takes an incoming OpenTelemetry trace, converts them into Sentry spans and transactions
+// and sends them using Sentry's transport.
+func (s *SentryExporter) pushTraceData(_ context.Context, td pdata.Traces) (droppedSpans int, err error) {
+	// For a ResourceSpan, InstrumentationLibrarySpan and Span struct if IsNil()
+	// is "true", all other methods will cause a runtime error.
+	resourceSpans := td.ResourceSpans()
+	if resourceSpans.Len() == 0 {
+		return 0, nil
+	}
+
+	maybeOrphanSpans := make([]*sentry.Span, 0, td.SpanCount())
+
+	// Maps all child span ids to their root span.
+	idMap := make(map[string]string)
+	// Maps root span id to a transaction.
+	transactionMap := make(map[string]*sentry.Event)
+
+	for i := 0; i < resourceSpans.Len(); i++ {
+		rs := resourceSpans.At(i)
+		if rs.IsNil() {
+			continue
+		}
+
+		resourceTags := generateTagsFromResource(rs.Resource())
+
+		ilss := rs.InstrumentationLibrarySpans()
+		for j := 0; j < ilss.Len(); j++ {
+			ils := ilss.At(j)
+			if ils.IsNil() {
+				continue
+			}
+
+			library := ils.InstrumentationLibrary()
+
+			spans := ils.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				otelSpan := spans.At(k)
+				if otelSpan.IsNil() {
+					continue
+				}
+
+				sentrySpan := convertToSentrySpan(otelSpan, library, resourceTags)
+
+				// If a span is a root span, we consider it the start of a Sentry transaction.
```

Comment:
I'm not familiar with the transaction concept it will be good if someone from Sentry reviews this part.

Reply:
We've reviewed this in our own repo. See the PRs from: https://github.com/getsentry/opentelemetry-collector-contrib/compare/master...getsentry:backup/sentryexporter

cc @rhcarvalho @bruno-garcia for another double check.